src/core/Form/TextInput/TextInput.tsx

```diff
@@ -151,40 +137,46 @@ class BaseTextInput extends Component<TextInputProps> {
           [textInputClassNames.fullWidth]: fullWidth,
         })}
       >
-        <LabelText
-          htmlFor={this.id}
-          labelMode={labelMode}
-          as="label"
-          optionalText={optionalText}
-        >
-          {labelText}
-        </LabelText>
-        <HintText id={this.hintTextId}>{hintText}</HintText>
-        <HtmlDiv className={textInputClassNames.inputElementContainer}>
-          <Debounce waitFor={this.props.debounce}>
-            {(debouncer: Function) => (
-              <HtmlInput
-                {...passProps}
-                id={this.id}
-                className={textInputClassNames.inputElement}
-                type={type}
-                placeholder={visualPlaceholder}
-                {...{ 'aria-invalid': status === 'error' }}
-                {...getDescribedBy()}
-                onChange={(event: ChangeEvent<HTMLInputElement>) => {
-                  if (propOnChange) {
-                    const eventValue = event.currentTarget.value;
-                    debouncer(propOnChange, eventValue);
-                  }
-                }}
-              />
-            )}
-          </Debounce>
-          {resolvedIcon && <Icon {...newIconProps} />}
-        </HtmlDiv>
-        <StatusText id={this.statusTextId} status={status}>
-          {statusText}
-        </StatusText>
+        <HtmlSpan className={textInputClassNames.styleWrapper}>
+          <LabelText
+            htmlFor={id}
+            labelMode={labelMode}
+            as="label"
+            optionalText={optionalText}
+          >
+            {labelText}
+          </LabelText>
+          <HintText id={this.hintTextId}>{hintText}</HintText>
+          <HtmlDiv className={textInputClassNames.inputElementContainer}>
+            <Debounce waitFor={this.props.debounce}>
+              {(debouncer: Function) => (
+                <HtmlInput
+                  {...passProps}
+                  id={id}
+                  className={textInputClassNames.inputElement}
+                  type={type}
+                  placeholder={visualPlaceholder}
+                  {...{ 'aria-invalid': status === 'error' }}
+                  {...getConditionalAriaProp('aria-describedby', [
+                    statusText ? this.statusTextId : undefined,
+                    hintText ? this.hintTextId : undefined,
+                    ariaDescribedBy,
+                  ])}
+                  onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                    if (propOnChange) {
+                      const eventValue = event.currentTarget.value;
+                      debouncer(propOnChange, eventValue);
+                    }
+                  }}
+                />
+              )}
+            </Debounce>
+            {resolvedIcon && <Icon {...newIconProps} />}
```

Comment:
If `const resolvedIcon = icon || iconProps?.icon;` is really needed it might make sense to give the typing manually as now it is `any`:
```tsx
const resolvedIcon: BaseIconKeys = icon || iconProps?.icon;
```


These `newIconProps` could be inlined if wanted:

```tsx
{resolvedIcon && <Icon {...{ iconProps, icon: resolvedIcon }} />}
```

Reply:
Good point. I like the inline version better, as I'm not a big fan of newThisAndThatProps kind of names. Changed it.